src/timer.c

```diff
@@ -53,6 +57,41 @@ static long long total_ticks;
  *            Value to write into the compare register
  */
 #define write_compare(x) asm volatile("mtc0 %0,$11\n\t nop \n\t" :  : "r" (x) )
+/**
+ * @brief Read the compare register.
+ *
+ * @return the compare value
+ */
+#define COMPARE_READ() ({ \
+    uint32_t x; \
+    asm volatile("mfc0 %0,$11":"=r"(x)); \
+    x; \
+})
+
+/* @brief Timer is the special overflow timer. */
+#define TF_OVERFLOW    0x40
+
+/* @brief Timer has been called once in this interrupt. */
+#define TF_CALLED      0x80
+
+/* @brief Update the compare register to match the first expiring timer. */
+static void timer_update_compare(timer_link_t *head) {
+    uint32_t now = TICKS_READ();
+    uint32_t smallest = 0xFFFFFFFF;
+
+    while (head)
+    {
+		uint32_t left = TICKS_DISTANCE(now, head->left);
+		if (left < smallest)
```

Comment:
No we cannot use a signed integer here. We need to calculate how many ticks are left before a timer triggers. If a timer triggers at 0 and now is 0x100, we need to calculate 0xFFFFFE00 which is a very large (unsigned) number, which is correct because there are 4294966784 cycles left before the counter becomes 0 again.

Reply:
Then using a plain difference (`head->left - now`) should also result in the same behavior and might be better for the reader considering the documentation (and resulting type) for `TICKS_DISTANCE`.

Reply:
Done